### PR TITLE
allow users to define a default priorityClass

### PR DESCRIPTION
**Pull request by narcis96**
Ref https://github.com/loft-sh/vcluster/issues/1718
Ref ENG-3563

### CODE REVIEW

File: chart/values.yaml [hunk 1]
```diff
@@ -457,6 +457,8 @@ controlPlane:
     overwriteManifests: ""
     # OverwriteConfig can be used to overwrite the coredns config
     overwriteConfig: ""
+    
+    priorityClassName: null
```
Comment (johannesfrey) [inline chart/values.yaml:461]: You have to add this field under [`config.go`](https://github.com/loft-sh/vcluster/blob/main/config/config.go#L937-L955) and [`values.yaml`](https://github.com/loft-sh/vcluster/blob/main/config/values.yaml#L255) and then run `go run ./hack/schema/main.go`

**Review by johannesfrey — CHANGES_REQUESTED** [review]

File: chart/values.yaml (hunk 1 above)
Comment (narcis96) [reply chart/values.yaml]: fixed

File: chart/templates/coredns-configmap.yaml [hunk 2]
```diff
@@ -152,7 +152,7 @@ data:
 {{ toYaml .Values.controlPlane.coredns.deployment.pods.labels | indent 12 }}
           {{- end }}
         spec:
-          priorityClassName: "system-cluster-critical"
+          priorityClassName: {{ .Values.controlPlane.coredns.priorityClassName | default "\"system-cluster-critical\"" }}
```
Comment (johannesfrey) [inline chart/templates/coredns-configmap.yaml:155]:
```suggestion
          priorityClassName: {{ .Values.controlPlane.coredns.priorityClassName | default "system-cluster-critical" | quote }}
```

File: config/config.go [hunk 3]
```diff
@@ -952,6 +952,9 @@ type CoreDNS struct {
 
 	// OverwriteManifests can be used to overwrite the coredns manifests used to deploy coredns
 	OverwriteManifests string `json:"overwriteManifests,omitempty"`
+
+	// PriorityClassName specifies the priority class name for CoreDNS pods.
```
Comment (johannesfrey) [inline config/config.go:956]:
```suggestion
	// PriorityClassName specifies the priority class name for the CoreDNS pods.
```

File: chart/values.schema.json [hunk 4]
```diff
@@ -535,6 +535,10 @@
         "overwriteManifests": {
           "type": "string",
           "description": "OverwriteManifests can be used to overwrite the coredns manifests used to deploy coredns"
+        },
+        "priorityClassName": {
```
Comment (johannesfrey) [inline chart/values.schema.json:539]: I guess this also must find it's way into the docs: https://github.com/loft-sh/vcluster-docs/tree/main/docs/_partials/config/controlPlane/coredns

There is [`gen-partials.sh`](https://github.com/loft-sh/vcluster-docs/blob/main/hack/gen-partials.sh) which updates this. Might be that this done for the next beta release anyway though?! (cc @aimeeu)

**Comment by heiko-braun** [comment]
@narcis96 
> Seems like you forgot to run 'go run hack/schema/main.go' before committing your changes!

File: chart/values.schema.json (hunk 4 above)
Comment (aimeeu) [reply chart/values.schema.json:539]: The process is manual and yes, we generate new partials for each release.

Comment (narcis96) [reply chart/values.schema.json:539]: so we should do it in another MR, right ?

**Comment by FabianKramm** [comment]
@narcis96 we should not have `system-critical` as a default anymore, its fine that we can configure it now, but it shouldn't be set by default

File: chart/templates/coredns-configmap.yaml [hunk 5]
```diff
@@ -152,7 +152,7 @@ data:
 {{ toYaml .Values.controlPlane.coredns.deployment.pods.labels | indent 12 }}
           {{- end }}
         spec:
-          priorityClassName: "system-cluster-critical"
+          priorityClassName: {{ .Values.controlPlane.coredns.priorityClassName | default "" | quote}}
```
Comment (FabianKramm) [inline chart/templates/coredns-configmap.yaml:155]: Just do `{{ .Values.controlPlane.coredns.priorityClassName | quote }}` here

**Review by FabianKramm — CHANGES_REQUESTED** [review]

File: chart/templates/coredns-configmap.yaml (hunk 5 above)
Comment (narcis96) [reply chart/templates/coredns-configmap.yaml]: done

File: chart/templates/coredns-configmap.yaml [hunk 6]
```diff
@@ -152,7 +152,7 @@ data:
 {{ toYaml .Values.controlPlane.coredns.deployment.pods.labels | indent 12 }}
           {{- end }}
         spec:
-          priorityClassName: "system-cluster-critical"
+          priorityClassName: {{ .Values.controlPlane.coredns.priorityClassName | quote}}
```
Comment (johannesfrey) [inline chart/templates/coredns-configmap.yaml:155]:
```suggestion
          priorityClassName: {{ .Values.controlPlane.coredns.priorityClassName | quote }}
```

**Review by johannesfrey — APPROVED** [review]

**Review by FabianKramm — APPROVED** [review]

**Pull request merged**